src/main/js/helpers/dateline.js

```diff
@@ -144,7 +164,32 @@ const createDateline = (scale, config, canvasSVG) => {
             )
         );
     });
-    translateDateline(scale, config, canvasSVG, getYAxisYPosition);
+    translateDateline(scale, config, canvasSVG, getYAxisYPosition, transition);
+
+    if (
+        config.settingsDictionary.shouldCreateDatelineDefs &&
+        config.dateline.length > 0
+    ) {
+        const shapeHeightArr = [];
+        const shape = d3.selectAll(`.${styles.datelinePoint}`);
+        shape[0].forEach((element) => {
```

Comment:
I am wondering if we can move this in `dateline.js`. We are essentially duplicating in Gantt, Graph and Timeline with some minor to no changes, arent we?

Reply:
Since both the gantt and graph uses different way to calculate the "getYAxisYPosition" that is why we can not make the function common for both graph and gantt.